x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_details/hooks/use_monitor_query_id.ts

```diff
@@ -5,11 +5,38 @@
  * 2.0.
  */
 
+import { useMemo } from 'react';
+import { UrlFilter } from '@kbn/exploratory-view-plugin/public';
+import { useSelectedLocation } from './use_selected_location';
 import { ConfigKey } from '../../../../../../common/runtime_types';
 import { useSelectedMonitor } from './use_selected_monitor';
 
-export const useMonitorQueryId = () => {
+export const useMonitorQueryId = (): {
+  monitorQueryId?: string;
+  queryIdFilter?: Record<string, string[]>;
+  locationFilter?: UrlFilter[];
+} => {
   const { monitor } = useSelectedMonitor();
+  const monitorQueryId = monitor?.[ConfigKey.MONITOR_QUERY_ID];
+  const selectedLocation = useSelectedLocation();
 
-  return monitor?.[ConfigKey.MONITOR_QUERY_ID];
+  return useMemo(() => {
+    if (!monitorQueryId || !selectedLocation) {
+      return {};
+    }
+    return {
+      monitorQueryId,
+      queryIdFilter: {
+        'monitor.id': [monitorQueryId],
+      },
+      locationFilter: [
+        {
+          field: 'observer.geo.name',
+          // in 8.6.0, observer.geo.name was mapped to the id,
+          // so we have to pass both values to maintain history
+          values: [selectedLocation.label, selectedLocation.id],
+        },
+      ],
```

Comment:
It's a good idea to abstract query and location filters this way. But would be nice to have a separate `useQueryFilters` hook for the purpose.